### PR TITLE
Integrate transport

### PR DESCRIPTION
**Pull request by narekhovhannisyan**
## Motivation

Integrate nodemailer transport into mailtrap-nodejs client.

## Changes

- Integrated transport.
- Extracted static vars to config.
- Added sample of transport usage.
- Updated documentation accordingly.

### CODE REVIEW

**Review by mklocek — APPROVED** [review]

File: package.json [hunk 1]
```diff
@@ -18,14 +18,17 @@
   "author": "Railsware Products Studio LLC",
   "license": "MIT",
   "dependencies": {
-    "axios": ">=0.27"
+    "axios": ">=0.27",
+    "nodemailer": "^6.9.4"
```
Comment (VladimirTaytor) [inline package.json:22]: I think we can add `nodemailer` as `peerDependencies`
- and maybe even mark it as optional using [peerDependenciesMeta](https://docs.npmjs.com/cli/v9/configuring-npm/package-json#peerdependenciesmeta)
- as most of our users would use SDK-as-SDK and people who needs `nodemailer-transporter` would have their own nodemailer
- #risk `declare` trick would not work if we will have "our" nodemailer and users would install their own

File: README.md [hunk 2]
```diff
@@ -147,12 +147,77 @@ client
...
   .catch(console.error);
 ```
 
+### Nodemailer Transport
+
+You can provide Mailtrap specific keys like `category`, `customVariables`, `templateUuid` and `templateVariables`.
+
+```ts
+import { readFileSync } from "fs";
+import Nodemailer from "nodemailer";
```
Comment (VladimirTaytor) [inline README.md:161]: We should make it clear that users would need to install `nodemailer` by themselves

Comment (narekhovhannisyan) [reply README.md:161]: @VladimirTaytor Will mention that in README.md file

File: package.json (hunk 1 above)
Comment (VladimirTaytor) [reply package.json]: Actually, as I'm thinking further...
- What would happen if the user would try to use our SDK without `nodemailer` installed?
  - Would it be even possible, as we are exporting `MailtrapTransport` from the index 🤔 
- Let's double check during `alfa` testing 👍

**Review by VladimirTaytor — APPROVED** [review]

**Review by VladimirTaytor — APPROVED** [review]

**Pull request merged**